docs/src/types.md

```diff
@@ -0,0 +1,128 @@
+
+None of the types below need to be constructed directly; they arise from
+performing operations between `LinearMap` objects or by calling the `LinearMap`
+constructor described next.
+
+### `LinearMap`
+
+Abstract supertype
+
+```@docs
+LinearMaps.LinearMap
+```
+
+### `FunctionMap`
+
+Type for wrapping an arbitrary function that is supposed to implement the
+matrix-vector product as a `LinearMap`; see above.
+
+### `WrappedMap`
+
+Type for wrapping an `AbstractMatrix` or `LinearMap` and to possible redefine
+the properties `isreal`, `issymmetric`, `ishermitian` and `isposdef`. An
+`AbstractMatrix` will automatically be converted to a `WrappedMap` when it is
+combined with other `AbstractLinearMap` objects via linear combination or
+composition (multiplication). Note that `WrappedMap(mat1)*WrappedMap(mat2)`
+will never evaluate `mat1*mat2`, since this is more costly than evaluating
+`mat1*(mat2*x)` and the latter is the only operation that needs to be performed
+by `LinearMap` objects anyway. While the cost of matrix addition is comparable
+to matrix-vector multiplication, this too is not performed explicitly since
+this would require new storage of the same amount as of the original matrices.
+
+### `ScaledMap`
+
+Type for representing a scalar multiple of any `LinearMap` type. A
+`ScaledMap` will be automatically constructed if real or complex `LinearMap`
+objects are multiplied by real or complex scalars from the left or from the
+right.
+
+### `UniformScalingMap`
+
+Type for representing a scalar multiple of the identity map (a.k.a. uniform
+scaling) of a certain size `M=N`, obtained simply as `LinearMap(λI, M)`,
+where `I` is the `LinearAlgreba.UniformScaling` object.
+The type `T` of the resulting `LinearMap` object is inferred from the type of
+`λ`. A `UniformScalingMap` of the correct size will be automatically
+constructed if `LinearMap` objects are multiplied by scalars from the left
+or from the right (respecting the order of multiplication), if the scalar `λ`
+is either real or complex.
+
+### `LinearCombination`, `CompositeMap`, `TransposeMap` and `AdjointMap`
+
+Used to add/multiply/transpose/adjoint `LinearMap` objects lazily, don't need to be constructed explicitly.
+
+```@docs
++(::LinearMap,::LinearMap)
+*(::LinearMap,::LinearMap)
+LinearAlgebra.transpose(::LinearMap)
+LinearAlgebra.adjoint(::LinearMap)
+```
+
+### `KroneckerMap` and `KroneckerSumMap`
+
+Types for representing Kronecker products and Kronecker sums, resp., lazily.
+
+```@docs
+Base.kron(::LinearMap,::LinearMap)
+LinearMaps.:⊗
+kronsum
+LinearMaps.:⊕
+```
+
+### `BlockMap` and `BlockDiagonalMap`
+
+Types for representing block (diagonal) maps lazily.
+
+```@docs
+Base.hcat
+Base.vcat
+Base.hvcat
+Base.cat
+SparseArrays.blockdiag
+```
+
+## Methods
+
+### Multiplication methods
+
+```@docs
+Base.:*(::LinearMap,::AbstractVector)
+Base.:*(::LinearMap,::AbstractMatrix)
+LinearAlgebra.mul!(::AbstractVector,::LinearMap,::AbstractVector)
```

Comment:
These work for `AbstractVecOrMat`

Reply:
Not yet as of with this commit. It's only after #97, so we need to remember to fix this there.